src/cljam/io/bam/decoder.clj

```diff
@@ -124,7 +124,10 @@
          cigar-bytes (lsb/read-bytes buffer (* n-cigar-op 4))
          [cigar ^long len] (cigar/decode-cigar-and-ref-length cigar-bytes)
          ref-end     (int ^long (if (zero? len) pos (dec (+ pos len))))
-         seq         (decode-seq (lsb/read-bytes buffer (quot (inc l-seq) 2)) l-seq)
+         seq         (if (zero? l-seq)
+                       "*"
+                       (decode-seq (lsb/read-bytes
+                                    buffer (quot (inc l-seq) 2)) l-seq))
```

Comment:
I’d prefer that `decode-seq` has the same level of granularity as `decode-qual`. If `decode-qual` returns `"*"` for an empty qual, users would expect `decode-seq` to return `"*"` for an empty seq.

With regard to code redundancy as well, I think it would be better to move the code dealing with `"*"` into `decode-seq` itself.